### DISCUSSION TITLE
Ideas

### DISCUSSION THREAD

**Discussion by danieldelcore** [blairjordan/codachi#24] [Ideas]
- Ability to click on your pet to interact with them (happiness?)
- Ability to feed them food, for them to sleep, and basic needs
- Lack of tending to their basic needs impacts XP progression 
- Keep the UI very minimal, instead of showing stats, show "speech bubbles with their requests".

![m5d1](https://github.com/blairjordan/codachi/assets/3030010/3b3f4008-c5c3-4191-acdd-b874238f58bb)

**Comment by blairjordan** [comment]
> Ability to click on your pet to interact with them (happiness?)
> Ability to feed them food, for them to sleep, and basic needs

I like these ideas. Thanks for posting! I have a couple of thoughts:

- Adding manual interactions would make the plugin more interesting, but could also make users feel compelled to interact with the plugin, which *could* become annoying 🙂

- I wonder if actions within regular VS Code could correspond to interactions, i.e.,
  - Writing code = feed (food drops from the sky for codachi to eat after N keystrokes)
  - Sleep = taking breaks from coding, idle time
The problem with this approach is that we are quite limited by inputs (i.e., keystrokes, timers) - Ideally I'd love to integrate with `git` (checkins/ PRs giving you XP instead of raw keystrokes).

I'm personally undecided between both approaches 😭

> Lack of tending to their basic needs impacts XP progression

There is a form of this in progress (although I haven't revisited it in some time).
The effect I had in mind was a "buff", i.e., coding velocity increases the speed in which XP accumulates.

https://github.com/blairjordan/codachi/pull/13

Again, the inputs for "tending to their basic needs" depends on the manual vs passive interactions above.

> Keep the UI very minimal, instead of showing stats, show "speech bubbles with their requests".

This is also what I had in mind for the UI 🙌 I think displaying nice speech bubbles will keep the style consistent.
We could also use icons in the speech bubbles to state needs without explicitly stating as text, e.g. food = 🍗, sleep = 💤, etc.
